Dev/adp5055

acatapan opened the pull request:
## Pull Request Description

This PR provides support for ADP5055, which combines three high performance buck regulators enabling direct
connection to high input voltages up to 18V with no pre-regulators. This controller integrates a high precision 8-bit digital-to-analog converter (DAC) to enable the output voltage dynamic voltage scaling (DVS) via the PMBus®-compatible, 2-wire interface. The PMBus interface provides other flexible configurations, such as start-up and shutdown sequence control, individual forced pulse-width modulation or power saving mode (FPWM or PSM) selection, an output discharge switch, and a power-good signal.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [X] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [X] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [X] I have performed a self-review of the changes
- [X] I have commented my code, at least hard-to-understand parts
- [X] I have build all projects affected by the changes in this PR
- [X] I have tested in hardware affected projects, at the relevant boards
- [X] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

kister-jimenez commented:
/AzurePipelines run

kister-jimenez commented:
/AzurePipelines run

kister-jimenez commented:
/AzurePipelines run

rccalam commented on drivers/power/adp5055/adp5055.h:
```diff
@@ -0,0 +1,255 @@
+/** Write command to ADP5055 device. */
+int adp5055_write(struct adp5055_desc *desc, uint8_t address, uint8_t data);
+
+int adp5055_read_status(struct adp5055_desc *desc, uint8_t *status_val);
```
Add doxygen comments to all API declarations

acatapan replied:
Hi @rccalam , added doxygen commits (rebased fixup to commit 1c1fc5b)

kister-jimenez commented:
/AzurePipelines run

buha commented on drivers/power/adp5055/adp5055.h:
```diff
@@ -0,0 +1,288 @@
+#define ADP5055_VID3_GO		NO_OS_BIT(2)
+
+#define ADP5055_EN_MODE_MASK	NO_OS_GENMASK(1, 0)
+#define ADP5055_DVS_AUTO		NO_OS_BIT(4)
```
mixed indentation, try to get them to allign using a tab equal to 8 spaces

buha commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,717 @@
+	int ret;
+	uint8_t data;
+
+	ret = adp5055_read(desc, ADP5055_CTRL_MODE1, &data);
```
create a static update bits function that does this read reg -> modify with mask -> write reg sequence and you can reuse that in all these functions as a one-liner

acatapan replied:
Hi @buha , added a static function: adp5055_update_register. Used this to replace all instances updating a register with a single mask

buha commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,717 @@
+		goto free_desc;
+
+	/* Time needed for the ADP5055 to power-up. */
+	no_os_mdelay(52);
```
this delay is useful only of you reset it by software or hardware (gpio) but i checked the datasheet and it doesn't have any such feature

so idk... there is no way to tell how much time passed since POR, keeping the line there doesn't do any good and removing it doesn't do any harm

acatapan replied:
Hi @buha , removed software delay

buha commented on drivers/power/adp5055/iio_adp5055.c:
```diff
@@ -0,0 +1,1636 @@
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "dvs_auto_enable",
```
i'd delete the _enable suffix in all these, it makes them longer than needed

acatapan replied:
Hi @buha , removed all _enable suffixes

buha commented on drivers/power/adp5055/iio_adp5055.c:
```diff
@@ -0,0 +1,1636 @@
+	if (!iio_adp5055->adp5055_desc)
+		return -EINVAL;
+
+	switch (channel->address) {
```
you have several instances where you can delete the whole switch and do something like:

```
reg = ADP5055_DVS_LIM1 + channel->address;
```

Just make sure channel->address is within 0 and 2.

acatapan replied:
Hi @buha , removed all switch channel instances to using the channel->address as offset

buha commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,717 @@
+
+	switch (ch) {
+	case ADP5055_VOUT1:
+		ch_offset = 0;
```
same here, you can drop the switch and do `ch_offset = ch;` and that's it

acatapan replied:
Hi @buha , removed all switch channel instances to using enum as is with added check if channel is within 0 and 2

buha commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,717 @@
+
+	switch (ch) {
+	case ADP5055_VOUT1:
+		mask = ADP5055_PSM1_ON;
```
this you can also simplify by dropping the whole switch and replacing it with `mask = ADP5055_PSM1_ON << ch;` and you can apply it in many places

acatapan replied:
Hi @buha , replaced all channel switch instances for bit masks to using the channel enum as bit shift of starting mask

buha commented on projects/adp5055/README.rst:
```diff
@@ -0,0 +1,176 @@
+
+.. code-block:: bash
+
+	EXAMPLE = basic
```
I think this needs to spell as `EXAMPLE=basic` without whitespace as whitespace will give:
```
$ make EXAMPLE = basic
make: *** empty variable name.  Stop.
```

acatapan replied:
Hi @buha , removed whitespaces in commit squashed to add project documentation

amiclaus reviewed (commented):
all you commits are lacking a body. please add one for each of them.

buha commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,548 @@
+			    enum adp5055_en_mode mode)
+{
+	int ret;
+	uint8_t data;
```
apply this to all the functions where you use the update function

buha replied:
if not enabled by default (although I'm pretty sure it is), use -Wunused-variable and the goal is to not have this kind of errors while compiling adp5055.c

acatapan replied:
Hi @buha , fixed all unused variables and always false conditions. I missed these warnings when I added the update function.

buha commented on drivers/power/adp5055/iio_adp5055.c:
```diff
@@ -0,0 +1,1439 @@
+};
+
+enum adp5055_iio_enable_type {
+	ADP5055_IIO_VID_ENABLE,
```
i'd remove the _ENABLE suffixes here as well

acatapan replied:
Hi @buha , removed the _ENABLE suffixes here too

buha commented on drivers/power/adp5055/iio_adp5055.c:
```diff
@@ -0,0 +1,1439 @@
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int adp5055_iio_read_enable(void *dev, char *buf, uint32_t len,
```
for it to make sense, this function would be better named `adp5055_iio_read_attr` with the `adp5055_iio_read_attr_available` pair or similar because attributes are not necessarily binary true/false switches, they could be integer numbers, rational numbers etc. so use the umbrella term

acatapan replied:
Hi @buha , this function specifically reads attributes that are for enabling/disabling features in the device. Will replacing this with `adp5055_iio_read_attr` not cause confusion that it does not read all attributes?

acatapan replied:
renamed this function to `adp5055_iio_read_enable_attr` , a little longer but specifies only enable/disable attributes. The available and write functions are also renamed

acatapan commented on projects/adp5055/src/platform/maxim/platform_src.mk:
```diff
@@ -0,0 +1,11 @@
+        $(PLATFORM_DRIVERS)/maxim_i2c.c       \
+        $(PLATFORM_DRIVERS)/maxim_uart.c      \
+        $(PLATFORM_DRIVERS)/maxim_uart_stdio.c
+ 
```
Hi @amiclaus , added a body to all commits

acatapan commented on projects/adp5055/src/examples/iio_example/example.mk:
```diff
@@ -0,0 +1,4 @@
+IIOD = y
+
+INCS += $(DRIVERS)/power/adp5055/iio_adp5055.h
+SRCS += $(DRIVERS)/power/adp5055/iio_adp5055.c
```
Hi @amiclaus , added a new line here

acatapan commented on drivers/power/adp5055/adp5055.c:
```diff
@@ -0,0 +1,548 @@
+int adp5055_set_enable_mode(struct adp5055_desc *desc,
+			    enum adp5055_en_mode mode)
+{
+	int ret;
```
Hi @buha , this is fixed along with all other unused variables

kister-jimenez commented:
/AzurePipelines run

kister-jimenez commented:
/AzurePipelines run

acatapan commented:
Changelogs:

- Initial Rev 0:
  - Added ADP5055 driver
  - Added ADP5055 IIO driver
  - Added ADP5055 driver documentation
  - Added ADP5055 project
  - Added ADP5055 project documentation
- Rev 1
  - Copyright year changed to 2025
  - ADP5055 project
    - Changed Makefile src and header includes to the standard
    - Removed platform_include.h which is covered in parameters.h
  - ADP5055 driver documentation
    - Added entry to driver sphinx source
  - ADP5055 project documentation
    - Added entry to project sphinx source
- Rev 2
  - Fixed astyle issues for ADP5055 driver, IIO driver, and project source files
- Rev 3
  - Added documentation for all API
- Rev 4
  - Fixed mixed indentation
  - Removed startup delay in initialization
  - Added update register static function replacing all instances of writing to a register with a single mask
  - Replaced all switch channel cases for addresses and masks with a one line channel offset or bit shift
- Rev 5
  - Added a body to all commits
  - Removed unused variables and always false conditions
  - Removed _enable suffixes
